rfd/0001-rfd.md

```diff
@@ -0,0 +1,385 @@
+---
+authors: JP Bruins Slot (jpbruinsslot@gmail.com)
+state: discussion
+discussion: https://github.com/minvws/nl-kat-coordination/pull/3002
+labels: process
+---
+
+# RFD 0001: Requests For Discussion
+
+## Introduction
+
+At the basis of this proposal is the idea that RFDs are a way to propose and
+discuss changes to the project. They are a way to document the design and
+implementation of a feature or architectural decisions, and to solicit feedback.
+An RFD is designed to allow for discussion of rough ideas, and additionally the
+documents become a permanent record for more established ones.
+
+Depending on the state of the RFD they may be quickly iterated on in a branch,
+discussed actively as part of a pull request to be merged, or commented upon
+after having been published. The level of depth of an RFD depends on the topic
+and urgency of the RFD. Some can be used as documentation of the current
+knowledge on a topic, while others ay be more urgent and require less rigor.
+
+This proposal aims to facilitate better decision-making. By documenting ideas,
+requirements, and subsequent discussions, decisions become more informed and
+data-driven. The RFD process enhances communication and collaboration by
+providing a clear, structured format for capturing and sharing information.
+
+Documenting RFDs in plain text files ensures they are easily searchable and can
+be tracked over time. The self-documenting nature of RFDs clarifies the
+reasoning behind architectural design choices, helping to maintain transparency
+and continuity throughout the project.
+
+## Proposal
+
+### Design
+
+#### When to use an RFD
+
+An RFD should be used when a change is proposed that is non-trivial. This could
+be a new feature, a change to the architecture, or a change to the process. The
+RFD should be used to document the proposal and to solicit feedback from the
+team. Dependent on the project the RFD process can be used for different types
+of changes. Some examples for this project include:
+
+- ...
+- ...
+- ...
+
+#### How to start the RFD process
+
+1.  Reserve an RFD number
+
+    Check the repository for the next available RFD number.
+
+2.  Create branch for your RFD
+
+    Create a branch named after the RFD number you wish to reserve.
+    If less than 4 digits, pad with zeros. Before creating the branch, check
+    if the branch already exists.
```

Comment:
For me it's still unclear what the pattern is, is it a:
- branch name with a padded number only?
- a branch name consisting of text and ending with a padded number?

Based on the pattern used in the command below I inferred that you meant option two. In any case, I would personally prefer option two as it can contain a descriptive title. Alternatively, if we really want to make this as short as possible, I think the pattern `rfd-<padded number>` would also make sense (and is easier parseable if we want to automate some part of the documentation for RFDs, or such as the table in the README file as explained below)

Reply:
Yes it is indeed option two, I'll make sure that the documentation resembles that.